homeassistant/components/totalconnect/__init__.py

```diff
@@ -23,31 +29,66 @@
     },
     extra=vol.ALLOW_EXTRA,
 )
+"""
+PLATFORMS = ["alarm_control_panel", "binary_sensor"]
+
 
-TOTALCONNECT_PLATFORMS = ["alarm_control_panel", "binary_sensor"]
+async def async_setup(hass: HomeAssistant, config: dict):
+    """Set up by configuration file."""
+    if DOMAIN not in config:
+        return True
 
+    hass.async_create_task(
+        hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": SOURCE_IMPORT}, data=config[DOMAIN],
+        )
+    )
 
-def setup(hass, config):
-    """Set up TotalConnect component."""
-    conf = config[DOMAIN]
+    return True
 
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
+    """Set up upon config entry in user interface."""
+    hass.data.setdefault(DOMAIN, {})
+
+    conf = entry.data
     username = conf[CONF_USERNAME]
     password = conf[CONF_PASSWORD]
 
-    client = TotalConnectClient.TotalConnectClient(username, password)
+    client = await hass.async_add_executor_job(
+        TotalConnectClient.TotalConnectClient, username, password
+    )
 
-    if client.token is False:
+    if not client.is_logged_in():
         _LOGGER.error("TotalConnect authentication failed")
         return False
 
-    hass.data[DOMAIN] = TotalConnectSystem(username, password, client)
+    hass.data[DOMAIN][entry.entry_id] = TotalConnectSystem(username, password, client)
```

Comment:
We don't seem to use username and password on this instance.

Reply:
Maybe we can drop this instance all together?

Reply:
Good point.  Sorry I blindly followed an example and didn't stop to think about it.